Preparatory changes for GHA workflow consolidations.

suo opened the pull request:
Stack from [ghstack](https://github.com/ezyang/ghstack):
* #74306
* __->__ #74305

This PR includes all the new files we are adding as part of the workflow consolidation work.

These changes are split out and to be landed first to maintain forward compatibilty of workflows.

suo committed:
Prepatory changes for GHA workflow consolidations.

This PR includes all the new files we are adding as part of the workflow consolidation work.

These changes are split out and to be landed first to maintain forward compatibilty of workflows.

[ghstack-poisoned]

malfet commented on .github/workflows/_mac-build.yml:
```diff
@@ -0,0 +1,113 @@
+        shell: bash
+        run: |
+          echo "${GITHUB_WORKSPACE}"
+          rm -rf "${GITHUB_WORKSPACE}"
```
Why not `sudo` here?

malfet reviewed (approved):
I've noticed we've duplicated `Checkout PyTorch` action 20+ times. Same for "Clean workspace" and "Display EC2 information"

Can we unify those somehow?

suo committed:
Update on "Prepatory changes for GHA workflow consolidations."


This PR includes all the new files we are adding as part of the workflow consolidation work.

These changes are split out and to be landed first to maintain forward compatibilty of workflows.

[ghstack-poisoned]

suo committed:
Update on "Prepatory changes for GHA workflow consolidations."


This PR includes all the new files we are adding as part of the workflow consolidation work.

These changes are split out and to be landed first to maintain forward compatibilty of workflows.

[ghstack-poisoned]

suo committed:
Update on "Prepatory changes for GHA workflow consolidations."


This PR includes all the new files we are adding as part of the workflow consolidation work.

These changes are split out and to be landed first to maintain forward compatibilty of workflows.

[ghstack-poisoned]

suo committed:
Update on "Prepatory changes for GHA workflow consolidations."


This PR includes all the new files we are adding as part of the workflow consolidation work.

These changes are split out and to be landed first to maintain forward compatibilty of workflows.

[ghstack-poisoned]

suo committed:
Update on "Preparatory changes for GHA workflow consolidations."


This PR includes all the new files we are adding as part of the workflow consolidation work.

These changes are split out and to be landed first to maintain forward compatibilty of workflows.

[ghstack-poisoned]

suo committed:
Update on "Preparatory changes for GHA workflow consolidations."


This PR includes all the new files we are adding as part of the workflow consolidation work.

These changes are split out and to be landed first to maintain forward compatibilty of workflows.

[ghstack-poisoned]

suo committed:
Update on "Preparatory changes for GHA workflow consolidations."


This PR includes all the new files we are adding as part of the workflow consolidation work.

These changes are split out and to be landed first to maintain forward compatibilty of workflows.

[ghstack-poisoned]

suo committed:
Update on "Preparatory changes for GHA workflow consolidations."


This PR includes all the new files we are adding as part of the workflow consolidation work.

These changes are split out and to be landed first to maintain forward compatibilty of workflows.

[ghstack-poisoned]

suo committed:
Update on "Preparatory changes for GHA workflow consolidations."


This PR includes all the new files we are adding as part of the workflow consolidation work.

These changes are split out and to be landed first to maintain forward compatibilty of workflows.

[ghstack-poisoned]

suo committed:
Update on "Preparatory changes for GHA workflow consolidations."


This PR includes all the new files we are adding as part of the workflow consolidation work.

These changes are split out and to be landed first to maintain forward compatibilty of workflows.

[ghstack-poisoned]

suo commented:
@pytorchbot merge this
